tests/gtestmusig2.c

```diff
@@ -0,0 +1,416 @@
+#include <gtest/gtest.h>
+extern "C" {
+#include "../src/libmusig2.h"
+#ifndef NR_SIGS
+#define NR_SIGS 3
+#endif
+#include "config.h"
+
+
+
+TEST (musig2, not_enough_signatures) {
+
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+
+
+    int i, j, k, l;
+    int ind;
+    secp256k1_pubkey pk_list[NR_SIGNERS];    // Signers' public key list
+    secp256k1_pubkey batch_list[NR_SIGNERS * V * NR_MESSAGES];   // Stores the batches of signers
+    musig2_context_sig mcs_list[NR_SIGNERS]; // Array that holds NR_SIGNERS musig2_context_sig
+
+    int err;
+
+    /**** Initialization ****/
+    for (i = 0; i < NR_SIGNERS; i++) {
+        /* Generate a keypair for the signer and get batch commitments. */
+        err = musig2_init_signer(&mcs_list[i], ctx, NR_MESSAGES);
+        ASSERT_EQ(err, 1);
+
+        /* Store the public key of the signer in pk_list */
+        err = secp256k1_keypair_pub(ctx, &pk_list[i], &mcs_list[i].keypair);
+        ASSERT_EQ(err, 1);
+
+        /* Store the batch commitments of the signer in batch_list */
+        l = 0; // the index of the signer's commitment list.
+        for (k = 0; k < NR_MESSAGES; k++) {
+            for (j = 0; j < V; j++, l++) {
+                ind = (NR_SIGNERS * V * k) + (j * NR_SIGNERS) + i; // the index for the list of collected batches.
+                err = secp256k1_keypair_pub(ctx, &batch_list[ind], mcs_list[i].comm_list[l]);
+                ASSERT_EQ(err, 1);
+            }
+        }
+    }
```

Comment:
To avoid code duplication (this appears in all tests), we could create a function `initialize_signers` that performs these operations, rather than writing it down for each test. WDYT?